src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,73 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private final Map<Class<?>, Object> instances = new HashMap<>();
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        Object clazzImplamentationsIntance = null;
+        Class<?> clazz = findImplementation(interfaceClazz);
+        Field[] declaredFields = interfaceClazz.getDeclaredFields();
+        if (!clazz.isAnnotationPresent(Component.class)) {
+            throw new RuntimeException("Injection failed, "
+                    + "missing @Component annotaion on the class" + Component.class.getName());
+        }
```

Comment:
you should write this 'if' on top of this method, because:
1. if class/interface hasn't `@Component` you will do redundant actions
2. imagine that interfaceClazz will be some interface which doesn't have Component annotation, in this case you put this in findImplementation method and there we'll get into this 'if':
`  if (interfaceClazz.isInterface()) {
            return interfaceImplementations.get(interfaceClazz);
        }`
which will return to us null (because we don't have such key), so
Class<?> clazz will be null and if we call some method to null we will have `NullPointerException();`